src/flask/blueprints.py

```diff
@@ -241,6 +244,35 @@ def register(self, app, options, first_registration=False):
                 endpoint="static",
             )
 
+        # Merge app and self dictionaries.
+        def merge_dict_lists(self_dict, app_dict):
+            """Merges self_dict into app_dict. Replaces None keys with self.name.
+            Values of dict must be lists.
+            """
+            for key, values in self_dict.items():
+                key = self.name if key is None else f"{self.name}"
```

Comment:
This line doesn't seem to make much sense. It's basically equivalent to `key = str(self.name`)`

Reply:
I changed it to `else key` for now, but I'm unsure if that's the right approach. Logically, I don't see a reason for changing the key if it exists, but the tests seem to pass either way, so I'd like to double check first. 
@davidism @ThiefMaster What do you think?

Reply:
The implementation is supposed to be: if None, use the blueprint name, otherwise prepend the blueprint name to the key.

```python
key = self.name if key is None else f"{self.name}.{key}"
```

Reply:
@davidism Thanks! I'll update that!